src/node_http_parser.cc

```diff
@@ -519,6 +519,28 @@ class Parser : public AsyncWrap {
     args.GetReturnValue().Set(ret);
   }
 
+  static void GetUpgrade(Local<String> property,
+                         const PropertyCallbackInfo<Value>& args) {
+    Parser* parser = Unwrap<Parser>(args.Holder());
+
+    Local<Value> ret = Boolean::New(
+        parser->env()->isolate(),
+        parser->parser_.upgrade);
+
+    args.GetReturnValue().Set(ret);
+  }
+
+  static void SetUpgrade(Local<String> property,
+                         Local<Value> value,
+                         const PropertyCallbackInfo<void>& args) {
+    Parser* parser = Unwrap<Parser>(args.Holder());
+
+    bool upgrade = value->BooleanValue();
+    parser->parser_.upgrade = upgrade;
```

Comment:
That doesn't look correct to me.  `http_parser.upgrade` is only supposed to be read, not written, by clients of libhttp_parser.